diff --git a/core/jni/android_hardware_Camera.cpp b/core/jni/android_hardware_Camera.cpp
index 17967e0..18b8991 100644
--- a/core/jni/android_hardware_Camera.cpp
+++ b/core/jni/android_hardware_Camera.cpp
@@ -385,7 +385,7 @@ void JNICameraContext::setCallbackMode(JNIEnv *env, bool installed, bool manualM
     }
 }
 
-static void android_hardware_Camera_setLongshot(JNIEnv *env, jobject thiz, jboolean enable)
+/*static void android_hardware_Camera_setLongshot(JNIEnv *env, jobject thiz, jboolean enable)
 {
     ALOGV("setLongshot");
 #if defined(QCOM_HARDWARE) && !defined(ICS_CAMERA_BLOB)
@@ -440,7 +440,7 @@ static void android_hardware_Camera_sendHistogramData(JNIEnv *env, jobject thiz)
       jniThrowException(env, "java/lang/RuntimeException", "set histogram mode failed");
      }
 #endif
- }
+ }*/
 void JNICameraContext::addCallbackBuffer(
         JNIEnv *env, jbyteArray cbb, int msgType)
 {
@@ -707,7 +707,7 @@ static void android_hardware_Camera_setHasPreviewCallback(JNIEnv *env, jobject t
     context->setCallbackMode(env, installed, manualBuffer);
 }
 
-static void android_hardware_Camera_setMetadataCb(JNIEnv *env, jobject thiz, jboolean mode)
+/*static void android_hardware_Camera_setMetadataCb(JNIEnv *env, jobject thiz, jboolean mode)
 {
     ALOGV("setMetadataCb: mode:%d", (int)mode);
     JNICameraContext* context;
@@ -723,7 +723,7 @@ static void android_hardware_Camera_setMetadataCb(JNIEnv *env, jobject thiz, jbo
     if (rc != NO_ERROR) {
         jniThrowException(env, "java/lang/RuntimeException", "set metadata mode failed");
     }
-}
+}*/
 
 static void android_hardware_Camera_addCallbackBuffer(JNIEnv *env, jobject thiz, jbyteArray bytes, int msgType) {
     ALOGV("addCallbackBuffer: 0x%x", msgType);
@@ -1017,7 +1017,7 @@ static JNINativeMethod camMethods[] = {
   { "native_takePicture",
     "(I)V",
     (void *)android_hardware_Camera_takePicture },
-  { "native_setHistogramMode",
+/*  { "native_setHistogramMode",
     "(Z)V",
      (void *)android_hardware_Camera_setHistogramMode },
   { "native_setMetadataCb",
@@ -1028,7 +1028,7 @@ static JNINativeMethod camMethods[] = {
      (void *)android_hardware_Camera_sendHistogramData },
  { "native_setLongshot",
      "(Z)V",
-      (void *)android_hardware_Camera_setLongshot },
+      (void *)android_hardware_Camera_setLongshot },*/
   { "native_setParameters",
     "(Ljava/lang/String;)V",
     (void *)android_hardware_Camera_setParameters },
